### ISSUE TITLE
Improve output of `genErrors`

### ISSUE THREAD

**Issue by WhatisRT** [IntersectMBO/formal-ledger-specifications#546]
- [ ] `genErrors` always generates fully qualified names. The output would be a lot more readable if we postprocess the string to remove those qualifiers.
- [ ] We could also give a number that indicates which failure happened, for easier identification.
- [ ] The output generates non-breaking spaces. We may want to replace them with regular spaces (but maybe that should be a downstream responsibility?)

**Comment by WhatisRT** [comment]
Note that that PR didn't fix any of those issues

**Comment by williamdemeo** [comment]
> Note that that PR didn't fix any of those issues

As for the first issue, the fully qualified names issue is not an issue if we avoid `genErrors` and use simpler error handling, like the simple function I implemented to find the bug causing the latest conformance test failure.  When I used `show` in that function, the names did not come out fully qualified.

Also, using a bespoke error generator function has the added benefit of allowing us to say which error occurred.

I know this isn't a scalable, automated approach, but it's useful when we know what we're looking for (as in the most recent case).

**Comment by WhatisRT** [comment]
Yeah, this is strictly an issue about `genErrors`, not about producing error messages in general. Of course a manual approach is going to be prettier, but it has the downside of being brittle: if you change the order of some clauses without updating the error messages for example, the error message will be misleading. This actually happened in the past which is why we came up with `genErrors` in the first place.
